### PR TITLE
Added shift scale capabilities to least squares fit 

### LINKED ISSUE #7907: Add options to transform x values used in LeastSquaresFit vectorpostprocessor
### Description of the enhancement or error report
Add shifting and/or scaling capability to LeastSquaresFit vector postprocessor.

### Rationale for the enhancement or information for reproducing the error
The x data being used in the polynomial fit needs to be shifted and scaled sometimes.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
I will add input parameters (shift and scale factors) to the least squares fit object.

### PR DESCRIPTION
**Pull request by hoffwm**
closes #7907

### CODE REVIEW

File: framework/include/vectorpostprocessors/LeastSquaresFit.h [hunk 1]
```diff
@@ -69,6 +69,12 @@ class LeastSquaresFit :
   /// The number of samples to be taken
   unsigned int _num_samples;
 
+  /// Values used to scale and or shift x and y data
+  Real _x_scale;
+  Real _x_shift;
+  Real _y_scale;
+  Real _y_shift;
+
```
Comment (dschwen) [inline framework/include/vectorpostprocessors/LeastSquaresFit.h]: Make those `const`. Use `///@{ ... ///@}` to make the Doxygen comment apply to all for variables.

File: framework/src/vectorpostprocessors/LeastSquaresFit.C [hunk 2]
```diff
@@ -101,23 +109,33 @@ LeastSquaresFit::execute()
...
   if (_x_values.size() == 0)
     mooseError("In LeastSquresFit size of data in x_values and y_values must be > 0");
-  PolynomialFit pf(_x_values, _y_values, _order, true);
-  pf.generate();
 
+  //Create a copy of _x_values that we can modify.
+  std::vector<Real> x_values(_x_values.begin(), _x_values.end());
+  std::vector<Real> y_values(_y_values.begin(), _y_values.end());
+
+  for (unsigned int i = 0; i < _x_values.size(); ++i)
```
Comment (dschwen) [inline framework/src/vectorpostprocessors/LeastSquaresFit.C]:
```
for (auto i = beginIndex(_x_values); ; i < _x_values.size(); ++i)
```

**Review by dschwen — DISMISSED** [review]

**Comment by bwspenc** [comment]
Besides the minor issues pointed out by @dschwen, this is good.  We want these changes.

File: framework/include/vectorpostprocessors/LeastSquaresFit.h [hunk 3]
```diff
@@ -69,6 +69,14 @@ class LeastSquaresFit :
   /// The number of samples to be taken
   unsigned int _num_samples;
 
+  /// Values used to scale and or shift x and y data
+  ///@{
```
Comment (dschwen) [inline framework/include/vectorpostprocessors/LeastSquaresFit.h]: I'm not entirely sure it works that way. I think it should be 
```
///@{ Values used to scale and or shift x and y data
...
///@}
```

i.e. comment inside the bracket

**Comment by permcody** [comment]
Yes, it is being worked on by Sebastian. We'll invalidate affected targets
when this is resolved.

**Review by permcody — APPROVED** [review]

**Pull request merged**